webdriver/tests/find_element_from_shadow_root/find.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from tests.support.asserts import assert_success
+
+
+def get_shadow_root(session, element):
+    return session.transport.send(
+        "GET", "session/{session_id}/element/{element_id}/shadow".format(
+            session_id=session.session_id,
+            element_id=element.id))
+
+
+def find_element(session, shadow_root, using, value):
+    return session.transport.send(
+        "POST", "session/{session_id}/shadow/{shadow_id}/element".format(
+            session_id=session.session_id,
+            shadow_id=shadow_root.id),
+        {"using": using, "value": value})
+
+
+@pytest.fixture
+def get_checkbox_dom(inline):
+    return inline("""
+        <style>
+            custom-checkbox-element {
+                display:block; width:20px; height:20px;
+            }
+        </style>
+        <custom-checkbox-element></custom-checkbox-element>
+        <script>
+            customElements.define('custom-checkbox-element',
+                class extends HTMLElement {
+                    constructor() {
+                            super();
+                            this.attachShadow({mode: 'open'}).innerHTML = `
+                                <div><input type="checkbox"/></div>
+                            `;
+                        }
+                });
+        </script>""")
+
+
+def test_find_element(session, get_checkbox_dom):
+    session.url = get_checkbox_dom
+    custom_element = session.find.css("custom-checkbox-element", all=False)
+    shadow_root = get_shadow_root(session, custom_element).value
+    response = find_element(session, shadow_root, "css", "input")
+    assert_success(response)
```

Comment:
Make sure to check that we got the correct element returned, eg. it should be a checkbox.

Also could you duplicate most of the tests from `find_element/find.py`? Would be great to also have tests for missing parameters, non-existing browsing contexts and such. That should be too much work to get the existing tests adapted. Also `user_prompts.py` would be important.

Reply:
Added additional tests, including user prompt tests, in subsequent commits.